lib/excoveralls/lcov.ex

```diff
@@ -0,0 +1,73 @@
+defmodule ExCoveralls.Lcov do
+  @moduledoc """
+  Generate lcov output for results.
+  """
+
+  @file_name "lcov.info"
+
+  @doc """
+  Provides an entry point for the module.
+  """
+  def execute(stats, options \\ []) do
+    generate_lcov(stats, Enum.into(options, %{})) |> write_file(options[:output_dir])
+
+    ExCoveralls.Local.print_summary(stats)
+  end
+
+  def generate_lcov(stats, _options) do
+    lcov = Enum.map(stats, fn stat -> generate_lcov_file(stat) end) |> Enum.join("\n")
+    lcov <> "\n"
+  end
+
+  def generate_lcov_file(stat) do
+    da =
+      stat.coverage
+      |> Enum.with_index(1)
+      |> Enum.filter(fn {k, _v} -> k != nil end)
+      |> Enum.map(fn {k, v} -> {Integer.to_string(v), Integer.to_string(k)} end)
+      |> Enum.map(fn {line, count} -> "DA:" <> line <> "," <> count end)
+
+    foundlines =
+      stat.coverage
+      |> Enum.filter(fn v -> v != nil end)
+
+    lf = foundlines |> Enum.count()
+    lh = foundlines |> Enum.filter(fn v -> v > 0 end) |> Enum.count()
+
+    lines =
+      ["TN:", "SF:" <> stat.name] ++
```

Comment:
As per this line, `TN:` is expected to be a line without file name? (though the test is defined as `TN:test/fixtures/test.ex\n`).

Reply:
Oh, Im sorry. I did removed look at [this](https://github.com/mitchhentges/lcov-rs/wiki/File-format#lcov-info-file-output-of-lcov-rs)  but forgot fix test.

Reply:
Thank you for the reply. It's not in a hurry, but I appreciate if you could open a new PR with this test fixed, when you have time (if the test pass, I'll be merging it).